### ISSUE TITLE
Need YouTube video player support in Ensemble

### ISSUE THREAD

**Issue by reubensjohn** [EnsembleUI/ensemble#655]
The Ensemble video Player widget plays if provided the video link. Since youtube doesn't expose actual links/ or provide short lived links out, its hard to play Youtube videos. Is there a way to get a Youtube player for Ensemble like this one in Flutter Pub 
https://pub.dev/packages/youtube_player_iframe

Can we wrap this widget as a youtube widget and make it available in Ensemble?

**Comment by kmahmood74** [comment]
@fullstackctp start by proposing a YAML configuration for this.

**Comment by fullstackctp** [comment]
```
EnsemblePlayer:
    link: 
    aspectRatio:
    fullScreen:
    AutoStart:

```

**Comment by fullstackctp** [comment]
```
YoutubePlayer:
    link:
    aspectRatio:
    autoStart:
```

**Comment by kmahmood74** [comment]
@fullstackctp please review the properties that the https://pub.dev/packages/youtube_player_iframe provides and expose most of them in YAML. Looks like you may have just looked at the short sample code they have there. 

This Youtube player has a number of properties such as mute, showControls, showFullscreenButton, loop etc. We need to expose them in yaml so the developers may utilize them. Also the link should be smart enough to just take the video id or the full link. 

Please go a bit deeper into the package!

**Comment by fullstackctp** [comment]
```
YoutubePlayer:
    link:
    showControls:
    showFullScreen:
    showVideoAnnotation:
    loop:
    startSeconds:
    autoStart:
    aspectRatio:
    videoList: (If you want multiple videos to be played in a single player)
    enableFullScreenOnDrag:
    autoFullScreen: (when user starts playing the video, it automatically opens up fullscreen)
  
```
In the package there are multiple controls that they provide. So to expose them as well, i will need to create different getters and methods as well eg

getters
metadata  - title, author, videoId, 
PlayerbackRate - 0.25, 0.5, 1, 1.25
PlayBackQuality

methods
mute(), unMute(), nextVideo(), previousVideo(), setLoop(bool value)

I tried out its functionality in the web example provided, there is a small bug. It doesn't load playlist on the basis of playlistId. If you provide a list of video_Id then it works, but just providing playlist id, it doesn't load it up. (The is mentioned in their repository [sarbagyastha/youtube_player_flutter/issues/865])

**Comment by kmahmood74** [comment]
looks great. Please go ahead. cc @reubensjohn

**Comment by kmahmood74** [comment]
@fullstackctp please update documentation and create a kitchen sink example

**Comment by kmahmood74** [comment]
when done, close this ticket

**Comment by reubensjohn** [comment]
Pretty cool, will be using this soon. When will this be available to use? When will it show up in the ensemble Go?

**Comment by kmahmood74** [comment]
most likely by tomorrow evening. I'll post it here

**Comment by reubensjohn** [comment]
Awesome. Devs will love ensemble for this

**Comment by kmahmood74** [comment]
this was fixed in #886 closing
